app/src/main/java/com/example/imdb/SearchFragment.kt

```diff
@@ -32,24 +34,126 @@ class SearchFragment : Fragment() {
         showMovies(movieList())
     }
 
-    private fun showMovies(movieList: List<Movie>){
-        if ( movieAdapter == null){
-            val picasso = Picasso.get()
-            movieAdapter = MovieAdapter(picasso,movieList){ movie ->
-
+    private fun showMovies(movieList: List<Movie>) {
+        val picasso = Picasso.get()
+        movieAdapter = MovieAdapter(picasso, movieList) { movie ->
+            val args = Bundle().apply {
+                putParcelable(ARG_MOVIE, movie)
             }
-            binding.movieList.adapter = movieAdapter
+            findNavController().navigate(R.id.item_to_details_movies_series, args)
 
         }
+        binding.movieList.adapter = movieAdapter
+
     }
 
+
     private fun movieList(): List<Movie> {
```

Comment:
I see you're creating the same list of movies from different places, you can avoid this by creating either a method inside the data class or by creating a helper class and then just invoking the list from here (and all the other places you need it)

Reply:
great advice!! I'm implementing it